cli/lib/nf3.mjs

```diff
@@ -255,35 +261,64 @@ class Nf3 {
   }
 
   /**
-  Method for signing and submitting an Ethereum transaction to the
+  Method for signing an Ethereum transaction to the
   blockchain.
   @method
   @async
   @param {object} unsignedTransaction - An unsigned web3js transaction object.
   @param {string} shieldContractAddress - The address of the Nightfall_3 shield address.
   @param {number} fee - the value of the transaction.
   This can be found using the getContractAddress convenience function.
+  @returns {Oject} Signed transaction.
+  */
+  async _signTransaction(unsignedTransaction, contractAddress, fee) {
+    let tx;
+    let signed;
+
+    await this.nonceMutex.runExclusive(async () => {
+      // estimate the gasPrice
+      const gasPrice = await this.estimateGasPrice();
+      // Estimate the gasLimit
+      const gas = await this.estimateGas(contractAddress, unsignedTransaction);
+
+      // if we don't have a nonce, we must get one from the ethereum client
+      if (!this.nonce) this.nonce = await this.web3.eth.getTransactionCount(this.ethereumAddress);
```

Comment:
It would be nice to test this on a _near_ real blockchain. According to _[Ethereum Book, Transactions chapter](https://github.com/ethereumbook/ethereumbook/blob/develop/06transactions.asciidoc)_: 
_"Be careful when using the getTransactionCount function for counting pending transactions, because you might run into some problems if you send a few transactions in a row."_
...
_"When you build an application that constructs transactions, it cannot rely on getTransactionCount for pending transactions. Only when the pending and confirmed counts are equal (all outstanding transactions are confirmed) can you trust the output of getTransactionCount to start your nonce counter. Thereafter, keep track of the nonce in your application until each transaction confirms."_

Reply:
We will certainly test this. But we already applied this solution in the past

Reply:
In any case, the reason we have this nonce control is exactly the one you posted. If minces don’t match, we take ours